code/unit_tests/material_tests.dm

```diff
@@ -0,0 +1,32 @@
+/datum/unit_test/crafting_recipes_shall_not_have_inconsistent_materials
+	name = "MATERIALS: Crafting Recipes Shall Not Have Inconsistent Materials"
+
+/datum/unit_test/crafting_recipes_shall_not_have_inconsistent_materials/start_test()
+	var/list/failed_designs = list()
+	var/list/passed_designs = list()
+	for(var/owner_mat in SSmaterials.materials_by_name)
+		var/material/mat_datum = SSmaterials.get_material_datum(owner_mat)
+		for(var/datum/stack_recipe/recipe in mat_datum.recipes)
+			var/obj/product = recipe.spawn_result()
+			var/failed
+			if(product.matter && (recipe.use_material || recipe.use_reinf_material))
+				if(recipe.use_material && product.matter[recipe.use_material] >= recipe.req_amount)
```

Comment:
Isn't matter in units like 2000 or something but `req_amount` is in sheets?

Reply:
Yeah that's meant to have a `/SHEET_MATERIAL_AMOUNT` on it.